app/scripts/directives/stlCourtsDatepicker.dir.js

```diff
@@ -0,0 +1,372 @@
+'use strict';
+/*
+    Usage: <stl-courts-datepicker date-format="some format string"></stl-courts-datepicker>
+    <stl-courts-datepicker date-field="ctrl.dob" date-valid="ctrl.dobValid" date-over-18="ctrl.dobOver18" date-format="MM/DD/YYYY"></stl-courts-datepicker>
+*/
+
+angular.module('yourStlCourts').directive('stlCourtsDatepicker', function($timeout,moment){
+  return{
+    restrict: 'E',
+    scope:{
+      dateField:'=',
+      dateValid:'=',
+      dateOver18:'='
+    },
+    templateUrl: 'views/stlCourtsDatepicker.html',
+    controller: 'StlCourtsDatepicker as ctrl',
+    link: function($scope,element,attrs){
```

Comment:
Are you referring to me adding events within the link $(element).on('click',function(){})  vs in in the HTML ng-click="ctrl.clicked()"?
Or were there other things I wasn't doing?